gpytorch/kernels/kernel.py

```diff
@@ -185,7 +186,8 @@ def has_lengthscale(self):
     @property
     def lengthscale(self):
         if self.has_lengthscale:
-            return self._param_transform(self.raw_lengthscale).clamp(self.eps, 1e5)
+            constraint = self._constraints["raw_lengthscale"]
+            return constraint.transform(self.raw_lengthscale).clamp(self.eps, 1e5)
```

Comment:
Why do this if above we have just set up the infrastructure to pass in lower and upper bounds into the constraint definition? I.e. we could use an interval here. I guess the issue is that you want to use softplus?